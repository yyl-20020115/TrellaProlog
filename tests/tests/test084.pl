:-initialization(main).

:- op(700, xfx, rplus).

:- dynamic(dummy/4).

goal_expansion(L rplus R, R + L).

dummy(A,B,C,D) :- B rplus A, D rplus C.

dummy(A,B) --> { B rplus A }.

dummy((Head --> Goals), Clause, _, _) :-
        list_goal(Goals, Body),
        expand_term((Head --> Body), Clause).

dummy(g(Goal), []) --> [{Goal}, !].

main :- listing(dummy/4).
