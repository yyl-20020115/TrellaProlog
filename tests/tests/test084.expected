dummy(A,B,C,D) :- A+B,C+D.
dummy(A,B,C,D) :- A+B,C=D.
dummy((A-->B),C,D,E) :- list_goal(B,F),expand_term((A-->F),C).
dummy(g(A),[],B,C) :- B=[{A},!|C].
